src/yearn.ts

```diff
@@ -124,6 +138,28 @@ export class Yearn<T extends ChainId> {
     this.simulation = new SimulationInterface(this, chainId, this.context);
     this.strategies = new StrategyInterface(this, chainId, this.context);
 
+    this.configureSignerWithAllowlist();
+
     this.ready = Promise.all([this.services.asset.ready]);
   }
+
+  configureSignerWithAllowlist() {
+    const validateTx = async (transaction: Deferrable<TransactionRequest>) => {
+      if (!this.services.allowList) {
+        return true;
+      }
+
+      const [to, data] = await Promise.all([transaction.to, transaction.data]);
+      return await this.services.allowList.validateCalldata(to, data);
+    };
+
+    const orig = JsonRpcSigner.prototype.sendTransaction;
+    JsonRpcSigner.prototype.sendTransaction = async function(transaction: Deferrable<TransactionRequest>) {
```

Comment:
We may want to do something similar with blocknative notifications

Reply:
cc @gotzenx @xgambitox I think we handle block native notifications on the FE